pfsimulator/clm/clm.F90

```diff
@@ -315,42 +318,50 @@ subroutine clm_lsm(pressure,saturation,evap_trans,topo,porosity,pf_dz_mult,istep
      !=== Initialize CLM and DIAG variables
      if (clm_write_logs==1) write(999,*) "Initialize CLM and DIAG variables"
      do t=1,drv%nch 
-        clm(t)%kpatch = t
-        call drv_clmini (drv, grid, tile(t), clm(t), istep_pf) !Initialize CLM Variables
-     enddo
-
-     !=== Initialize the CLM topography mask 
-     !    This is two components: 
-     !    1) a x-y mask of 0 o 1 for active inactive and 
-     !    2) a z/k mask that takes three values 
-     !      (1)= top of LS/PF domain 
-     !      (2)= top-nlevsoi and 
+        clm%kpatch = t
+
+        !=== Initialize the CLM topography mask  @RMM  moved up from loop below

```

Comment:
In PF there is a top and a bottom (bottom is in a pending PR) 2D array with the z index of top/bottom.

Vector        *top = ProblemDataIndexOfDomainTop(problem_data);

I think those would eliminate the need for this loop?   Should we add that as an issue.   Bottom is present but once that is done I think we can use those directly.

Reply:
Created issue to address later.